packages/ui/useScroll/useScroll.js

```diff
@@ -818,27 +932,29 @@ const useScrollBase = (props) => {
 
 	// overscroll effect
 
-	function getEdgeFromPosition (position, maxPosition) {
-		if (position <= 0) {
-			return 'before';
-		} else if (position >= maxPosition - epsilon) {
-			return 'after';
-		} else {
-			return null;
-		}
-	}
+	const getOverscrollStatus = useCallback((orientation, edge) => {
+		return (mutableRef.current.overscrollStatus[orientation][edge]);
+	}, []);
 
-	function setOverscrollStatus (orientation, edge, overscrollEffectType, ratio) {
-		const status = mutableRef.current.overscrollStatus[orientation][edge];
-		status.type = overscrollEffectType;
-		status.ratio = ratio;
+	function applyOverscrollEffect (orientation, edge, overscrollEffectType, ratio = overscrollDefaultRatio) {
```

Comment:
It looks that this function could be a function variable with `useCallback`. Do I miss something?

Reply:
This PR is rather meant to be an attempt to solve the task, but I do not consider it complete. I got somewhat stuck and I would very much appreciate your input.
For this particular function, if I wrap it in useCallback I get this warning: "React Hook useCallback has a missing dependency: 'props'. Either include it or remove the dependency array. However, 'props' will change when *any* prop changes, so the preferred fix is to destructure the 'props' object outside of the useCallback call and refer to those specific props inside useCallback". 
I chose not to do anything for the time being so that it does not affect the component's functionality. What would you suggest I do when getting this warning? (I get it for a lot of other functions if I wrap them in useCallback). Thank you!

Reply:
I see what is your concern. I'll look into it. (No good idea at a glance, though)